apple/internal/apple_framework_import.bzl

```diff
@@ -223,6 +236,17 @@ def _apple_static_framework_import_impl(ctx):
     if ctx.attr.weak_sdk_frameworks:
         objc_provider_fields["weak_sdk_framework"] = depset(ctx.attr.weak_sdk_frameworks)
 
+    imported_swiftmodules = [
+        f
+        for f in header_imports
+        if _is_swiftmodule(f.basename)
+    ]
+
+    if imported_swiftmodules:
+        swift_toolchain = ctx.attr._toolchain[SwiftToolchainInfo]
+        swift_linkopts = swift_common.swift_runtime_linkopts(True, swift_toolchain)
+        objc_provider_fields["linkopt"] = depset(swift_linkopts)
```

Comment:
I think @keith 's right and we need at least the `SwiftUsageInfo` provider so that this [check](https://github.com/bazelbuild/rules_apple/blob/master/apple/internal/partials/swift_dylibs.bzl#L121) is true and we consider the binary that's linking against the framework to be considered for Swift stdlib dylib processing. 

I also think that returning this provider will also add the correct linkopts, since they're added using the same check.

So I'd change this block to something like:

```
# If we find any Swift module within the imported headers, add the SwiftUsageInfo provider
# to signal the top level targets that there is Swift involved in the build.
for file in header_imports:
    if _is_swift_module(f.basename):
        toolchain = ctx.attr._toolchain[SwiftToolchainInfo]
        providers.append(SwiftUsageInfo(toolchain = toolchain))
        break
```

that way we don't iterate over all files unnecesarily, and we let the downstream logic to add the required linkopts. Can you try this out and verify it fits your needs?

Reply:
I didn't know about `SwiftUsageInfo`, thanks.

Reply:
Updated.